src/rpg_sprites/Sprite_Character.js

```diff
@@ -203,7 +203,13 @@ Sprite_Character.prototype.createHalfBodySprites = function() {
 Sprite_Character.prototype.updatePosition = function() {
     this.x = this._character.screenX();
     this.y = this._character.screenY();
-    this.z = this._character.screenZ();
+
+    var newZ = this._character.screenZ();
+    if (newZ !== this.z) {
```

Comment:
I had first added a check on Game_CharacterBase.updateMove, but I guess keeping both conditions in the same class would make more sense.
I'll make a new commit changing it later.